libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -1267,9 +1267,12 @@ static void serializeConnectPacket( const MQTTConnectInfo_t * const pConnectInfo
         pIndex = encodeString( pIndex,
                                pWillInfo->pTopicName,
                                pWillInfo->topicNameLength );
-        pIndex = encodeString( pIndex,
-                               pWillInfo->pPayload,
-                               ( uint16_t ) pWillInfo->payloadLength );
+        if( pWillInfo->pPayload != NULL )
```

Comment:
please add a comment here why we have to add this check instead of a parameter  validation.

Reply:
Now that I think about it, this is incorrect since the length of the payload needs to be encoded in the next two bytes. Moved the check to inside `encodeString`, before the `memcpy` is called.

Reply:
Yes. That is true. Length of 0 has to be added. We just need to skip the memcpy.